### PR TITLE
Converts observation events, posters, and psionics to use decls instead of datums.

### PR DESCRIPTION
**Pull request by MarinaGryphon**
* /datum/observ -> /decl/observ
* /datum/poster -> /decl/poster
* /datum/psionic_faculty -> /decl/psionic_faculty
* /datum/psionic_power -> /decl/psionic_power
* Removes outdated debug code for observation events that's incompatible with decls.

### CODE REVIEW

**Comment by MarinaGryphon** [comment]
!review

File: code/__defines/misc.dm [hunk 1]
```diff
@@ -467,3 +467,9 @@ Define for getting a bitfield of adjacent turfs that meet a condition.
 #define COOK_CHECK_EXTRA	0
 #define COOK_CHECK_EXACT	1
 
+// Macros for observation events
+#define RAISE_EVENT(e, a...) decls_repository.get_decl(/decl/observ/##e).raise_event(##a)
```
Comment (skull132) [inline code/__defines/misc.dm:471]: I don't think `##a` is needed here. I used it in the first argument because it acts as a literal paste.

**Review by skull132 — CHANGES_REQUESTED** [review]

**Review by NonQueueingMatt — APPROVED** [review]
Looks fine other than what Skull said.

**Review by JohnWildkins — APPROVED** [review]

**Review by skull132 — APPROVED** [review]

**Comment by MarinaGryphon** [comment]
!wip

**Comment by NonQueueingMatt** [comment]
dead author

**Pull request closed without merging**